spiped/dispatch.c

```diff
@@ -132,11 +163,34 @@ callback_gotconn(void * cookie, int s)
 	if ((sas = sock_addr_duplist(A->sas)) == NULL)
 		goto err1;
 
+	/* Create new conn_list_node. */
+	if ((node_new = malloc(sizeof(struct conn_list_node))) == NULL)
+		goto err2;
+	node_new->prev = NULL;
+	node_new->next = NULL;
+
 	/* Create a new connection. */
-	if (proto_conn_create(s, sas, A->decr, A->nofps, A->requirefps,
-	    A->nokeepalive, A->K, A->timeo, callback_conndied, A) == NULL) {
+	if ((node_new->conn_cookie = proto_conn_create(s, sas, A->decr,
+	    A->nofps, A->requirefps, A->nokeepalive, A->K, A->timeo,
+	    callback_conndied, A)) == NULL) {
 		warnp("Failure setting up new connection");
-		goto err2;
+		goto err3;
+	}
+
+	/* Add conn_list_node node_new to accept_state A. */
+	if (A->conn_cookies == NULL) {
+		/* Nothing in list yet. */
+		A->conn_cookies = node_new;
+	} else {
+		/* Find end of list. */
```

Comment:
Good idea!  Also, that way we don't need to have any special case for the list not existing yet.